### ISSUE TITLE
Bisect smarter

### ISSUE THREAD

**Issue by AlexDaniel** [perl6/whateverable#195]
There are many additional things we can take into account when picking the next revision to test. Imagine this situation:
* Some code used to hang.
* It was fixed a few days ago, so it is relatively close to HEAD.
* We bisect this code with old=2015.12 new=HEAD (which is currently the default)
* Bisectable does a pre-test on starting points to determine bisect strategy. The test on HEAD takes 0.2 seconds, but on 2015.12 it takes 10 seconds (waiting for the process to be killed).
* Therefore, all further tests will take 10 seconds, until it narrows it down enough (it simply takes the commit in the middle of the remaining range, which is less than optimal in this case).

In other words, the “middle point” should be weighted by time it takes to run something on that side of the range. This way we will be able to bisect less efficiently in terms of the number of commits tested, but faster in terms of wall clock time.
